version: "3.7"

services:
  nginx:
    image: nginx:1.19.5
    ports:
      - 80:80
    networks:
      - dabin_network
    volumes:
      - /home/kangdb123-/nginx.conf:/etc/nginx/nginx.conf
      - dabin_static:/data/static/
      - dabin_media:/data/media/
      - gis_log:/var/log/nginx

  goaccess:
    image: allinurl/goaccess:1.5.2
    volumes:
      - gis_log:/var/log/nginx/
    ports:
      - 7890:7890
    command: /var/log/nginx/gis_access.log -o /var/log/nginx/report.html --log-format=COMBINED --real-time-html

  django:
    image: dabin_django:5
    networks:
      - dabin_network
    volumes:
      - dabin_static:/home/gis_web_mine/staticfiles/
      - dabin_media:/home/gis_web_mine/media/
    secrets:
      - DJANGO_SECRET_KEY
      - MARIADB_USER
      - MARIADB_PASSWORD

  mariadb:
    image: mariadb:10.5
    networks:
      - dabin_network
    volumes:
      - gis_mariadb:/var/lib/mysql
    secrets:
      - MARIADB_ROOT_PASSWORD
      - MARIADB_USER
      - MARIADB_PASSWORD
    environment:
      MARIADB_DATABASE: django
      MARIADB_USER_FILE: /run/secrets/MARIADB_USER
      MARIADB_PASSWORD_FILE: /run/secrets/MARIADB_PASSWORD
      MARIADB_ROOT_PASSWORD_FILE: /run/secrets/MARIADB_ROOT_PASSWORD


networks:
  dabin_network:

volumes:
  dabin_static:
  dabin_media:
  gis_mariadb:
  gis_log:

secrets:
  DJANGO_SECRET_KEY:
    external: true
  MARIADB_ROOT_PASSWORD:
    external: true
  MARIADB_USER:
    external: true
  MARIADB_PASSWORD:
    external: true